scheduler/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/mailbadger/app/emails"
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/queue"
+	"github.com/mailbadger/app/storage"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	driver := os.Getenv("DATABASE_DRIVER")
+	conf := storage.MakeConfigFromEnv(driver)
+	s := storage.New(driver, conf)
+
+	now := time.Now()
+	err := job(context.Background(), s, now)
+	if err != nil {
+		logrus.WithField("time", now).WithError(err).Error("failed to start campaign scheduler job")
+	}
+	end := time.Since(now)
+
+	logrus.Infof("Scheduler started at %v and took %f minutes to finish", now, end.Minutes())
+
+}
+
+func job(c context.Context, s storage.Storage, time time.Time) error {
+	scheduledCampaigns, err := s.GetScheduledCampaigns(time)
+	if err != nil {
+		return fmt.Errorf("failed to get scheduled campaigns: %w", err)
+	}
+
+	for _, cs := range scheduledCampaigns {
+		u, err := s.GetUser(cs.UserID)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"campaign_id": cs.CampaignID,
+				"user_id":     cs.UserID,
+			}).WithError(err).Error("failed to get user.")
+			continue
+		}
+		campaign, err := s.GetCampaign(u.ID, cs.CampaignID)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"campaign_id": cs.CampaignID,
+				"user_id":     cs.UserID,
+			}).WithError(err).Error("failed to get campaign.")
+			continue
+		}
+		if campaign.Status != entities.StatusDraft {
+			logrus.WithFields(logrus.Fields{
+				"campaign_id": cs.CampaignID,
+				"user_id":     cs.UserID,
+			}).WithError(err).Warn("campaign status is not draft.")
+			continue
+		}
+
+		template, err := storage.GetTemplate(c, campaign.BaseTemplate.ID, u.ID)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"campaign_id": cs.CampaignID,
+				"user_id":     cs.UserID,
+				"template_id": campaign.BaseTemplate.ID,
+			}).WithError(err).Error("failed to get template.")
+			continue
+		}
+		var templateData map[string]string
+		err = json.Unmarshal(cs.DefaultTemplateData, &templateData)
```

Comment:
Maybe we should have a `GetDefaultTemplateData` and `GetSegmentIDs` methods, for unmarshaling this data, like we have in the subscriber entity: https://github.com/mailbadger/app/blob/0313e359b6ebac6001f41682920c43a26e790ce2/entities/subscriber.go#L29